File: mobile/src/main/java/com/example/steven/stlbusarrivals/UI/Adapter/RouteSearchAdapter.java
package com.example.steven.stlbusarrivals.UI.Adapter;

import android.annotation.SuppressLint;
import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ArrayAdapter;
import android.widget.TextView;

import com.example.steven.stlbusarrivals.Model.Route;
import com.example.steven.stlbusarrivals.R;

import java.util.ArrayList;

/**
 * Created by Steven on 2016-01-26.
 */
public class RouteSearchAdapter extends ArrayAdapter<Route> {

    private LayoutInflater inflater;
    private Context context;

    public RouteSearchAdapter(Context context, int rowLayoutResourceId, ArrayList<Route> list) {
        super(context, rowLayoutResourceId, list);
        this.inflater = LayoutInflater.from(context);
        this.context = context;
    }

    @SuppressLint("DefaultLocale")
    @Override
    public View getView(int position, View view, ViewGroup parent) {

        ViewHolder holder;
        if (view != null) {
            holder = (ViewHolder) view.getTag();
        } else {
            view = inflater.inflate(R.layout.row_route_search_list, parent, false);
            holder = new ViewHolder();
            holder.routeName = (TextView) view.findViewById(R.id.tv_row_route_name);
            view.setTag(holder);
        }

        Route item = getItem(position);
        holder.routeName.setText(item.getName());

        return view;
    }

    static class ViewHolder {
        TextView routeName;
    }
}

File: mobile/src/main/java/com/tilly/steven/stlbusarrivals/ui/adapter/RouteSearchAdapter.kt
package com.tilly.steven.stlbusarrivals.ui.adapter

import android.annotation.SuppressLint
import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ArrayAdapter
import android.widget.TextView
import com.tilly.steven.stlbusarrivals.R
import com.tilly.steven.stlbusarrivals.model.Route
import java.util.*

/**
 * Created by Steven on 2016-01-26.
 */
class RouteSearchAdapter(ctx: Context, rowLayoutResourceId: Int, list: ArrayList<Route>) : ArrayAdapter<Route>(ctx, rowLayoutResourceId, list) {

    private val inflater: LayoutInflater = LayoutInflater.from(context)

    @SuppressLint("DefaultLocale")
    override fun getView(position: Int, view1: View?, parent: ViewGroup): View {
        var view = view1

        val holder: ViewHolder
        if (view != null) {
            holder = view.tag as ViewHolder
        } else {
            view = inflater.inflate(R.layout.row_route_search_list, parent, false)
            holder = ViewHolder()
            holder.routeName = view!!.findViewById<View>(R.id.tv_row_route_name) as TextView
            view.tag = holder
        }

        val item = getItem(position)
        holder.routeName.text = item!!.name

        return view
    }

    internal class ViewHolder {
        lateinit var routeName: TextView
    }
}
